tests/api_connexion/endpoints/test_role_and_permission_endpoint.py

```diff
@@ -146,3 +168,224 @@ def test_should_raise_403_forbidden(self):
+        )
+        assert response.status_code == 200
+        assert response.json['name'] == expected_name
+        assert response.json["actions"] == expected_actions
+
+    @parameterized.expand(
+        [
+            (
+                "?update_mask=name",
+                {
+                    "name": "mytest2",
+                    "actions": [{"resource": {"name": "Connections"}, "action": {"name": "can_create"}}],
+                },
+                "mytest2",
+                [],
+            ),
+            (
+                "?update_mask=name, actions",  # both name and actions in update mask
+                {
+                    "name": "mytest2",
+                    "actions": [{"resource": {"name": "Connections"}, "action": {"name": "can_create"}}],
+                },
+                "mytest2",
+                [{"resource": {"name": "Connections"}, "action": {"name": "can_create"}}],
+            ),
+        ]
+    )
+    def test_patch_should_respond_200_with_update_mask(
+        self, update_mask, payload, expected_name, expected_actions
+    ):
+        role = create_role(self.app, "mytestrole")
+        assert role.permissions == []
+        response = self.client.patch(
+            f"/api/v1/roles/{role.name}{update_mask}",
+            json=payload,
+            environ_overrides={'REMOTE_USER': "test"},
+        )
+        assert response.status_code == 200
+        assert response.json['name'] == expected_name
+        assert response.json['actions'] == expected_actions
+
+    def test_patch_should_respond_400_for_invalid_fields_in_update_mask(self):
+        role = create_role(self.app, "mytestrole")
+        payload = {"name": "testme"}
+        response = self.client.patch(
+            f"/api/v1/roles/{role.name}?update_mask=invalid_name",
+            json=payload,
+            environ_overrides={'REMOTE_USER': "test"},
+        )
+        assert response.status_code == 400
+        assert response.json['detail'] == "'invalid_name' in update_mask is unknown"
+
+    @parameterized.expand(
+        [
+            (
+                {
+                    "name": "testme",
+                    "permissions": [  # Using permissions instead of actions should raise
+                        {"resource": {"name": "Connections"}, "action": {"name": "can_create"}}
+                    ],
+                },
+                "{'permissions': ['Unknown field.']}",
+            ),
+            (
+                {
+                    "name": "testme",
+                    "actions": [
+                        {
+                            "view_menu": {"name": "Connections"},  # Using view_menu instead of resource
+                            "action": {"name": "can_create"},
+                        }
+                    ],
+                },
+                "{'actions': {0: {'view_menu': ['Unknown field.']}}}",
+            ),
+        ]
+    )
```

Comment:
We should also test what happens if the "structure" is valid, but a resource or action are invalid (i.e. if you have ` {"resource": {"name": "FooBars"}`)